ci/jenkinsfiles/common-lib.groovy

```diff
@@ -0,0 +1,241 @@
+  String preid = 'rc'
+  String nextPromotion = readMavenPom().getVersion().replace('-SNAPSHOT', '')
+  String version = "${nextPromotion}-${preid}.0" // first version ever
+
+  // find the latest tag if any
+  sh "git fetch origin 'refs/tags/v${nextPromotion}*:refs/tags/v${nextPromotion}*'"
+  String tag = sh(returnStdout: true, script: "git tag --sort=taggerdate --list 'v${nextPromotion}*' | tail -1 | tr -d '\n'")
+  if (tag) {
+    container('maven') {
+      version = sh(returnStdout: true, script: "npx --ignore-existing semver -i prerelease --preid ${preid} ${tag} | tr -d '\n'")
+    }
+  }
+  return version
+}
+
+String getPullRequestVersion() {
+  return "${readMavenPom().getVersion()}-${BRANCH_NAME}"
+}
+
+void gitCheckout(String reference) {
+  sh "git checkout ${reference}"
+}
+
+void gitCommit(String message, String option) {
+  echo '''
+    ----------------------------------------
+    Git Commit
+    ----------------------------------------
+  '''
+  sh """
+    git commit ${option} -m "${message}"
+  """
+}
+
+void gitPush(String reference) {
+  sh "git push origin ${reference}"
+}
+
+void gitTag(String tagname, String message) {
+  sh """
+    git tag -a ${tagname} -m "${message}"
+  """
+}
+
+def isPullRequest() {
+  return "${BRANCH_NAME}" =~ /PR-.*/
+}
+
+void lint() {
+  echo '''
+    ----------------------------------------
+    Run Linting Validations
+    ----------------------------------------
+  '''
+  sh """
+    cd ${FRONTEND_FOLDER}
+    npm run lint
+  """
+}
+
+String mavenArgs() {
+  String args = '-DskipITs=true -fae -B -nsu'
+  return args
+}
+
+boolean needsSaucelabs() {
+  return this.isPullRequest() && pullRequest.labels.contains('saucelabs')
+}
+
+void runFrontEndUnitTests() {
+  return {
+    stage('Run Unit tests: FrontEnd') {
+      container('maven') {
+        String context = 'coldstorage/utests/frontend'
+        String message = 'Unit tests - FrontEnd'
+        echo '''
+          ----------------------------------------
+          Run FrontEnd Unit tests
+          ----------------------------------------
+        '''
+        script {
+          this.setGitHubBuildStatus("${context}", "${message}", 'PENDING', "${GIT_URL}")
+          String sauceAccessKey = ''
+          String sauceUsername = ''
+          if (this.needsSaucelabs()) {
+            String saucelabesSecretName = 'saucelabs-coldstorage'
+            sauceAccessKey =
+              sh(script: "jx step credential -s ${saucelabesSecretName} -k key", returnStdout: true).trim()
+            sauceUsername =
+              sh(script: "jx step credential -s ${saucelabesSecretName} -k username", returnStdout: true).trim()
+          }
+          try {
+            withEnv(["SAUCE_USERNAME=${sauceUsername}", "SAUCE_ACCESS_KEY=${sauceAccessKey}"]) {
+              container('playwright') {
+                sh """
+                  cd ${FRONTEND_FOLDER}
+                  npm clean-install
+                  npm run test
```

Comment:
As discussed previously, this is forcing us to download all the dependencies again.
Guess we can move now with this, but probably creating a ticket to think further about it could be a good idea. (cc @RSalem07 )